package/web-component-util.ts

```diff
@@ -0,0 +1,89 @@
+const getStylesRecursively = (component) => {
+  const customElementStyles = []
+
+  if (component.styles) {
+    customElementStyles.push(...component.styles)
+  }
+
+  const childComponents = component.components
+  if (childComponents) {
+    Object.keys(childComponents).forEach((name) => {
+      const styles = getStylesRecursively(childComponents[name])
+      customElementStyles.push(...styles)
+    })
+  }
+
+  return customElementStyles
+}
+
+const nearestElement = (el) => {
+  while (el?.nodeType !== 1 /* ELEMENT */) {
+    if (!el.parentElement) {
+      throw new Error(
+        'No parent element found, the rootComponent must be wrapped in a HTML element (e.g. <template><div> app content </div></template>)'
+      )
+    }
+    el = el.parentElement
+  }
+  return el
+}
+
+const getCustomElementStyle = (styles) => {
```

Comment:
Not sure what the usage of `getCustomElementStyle`

Reply:
I will remove it in the next commit.